server/views_test.go

```diff
@@ -52,3 +56,79 @@ func TestHandlePlayersList(t *testing.T) {
 
 	assertStatusOk(rr, t)
 }
+
+func TestHandleAddPlayerGet(t *testing.T) {
+	testCtx := makeTestCtx(t)
+	defer testCtx.mockCtrl.Finish()
+
+	req, err := http.NewRequest("GET", "/players/add", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	rr := httptest.NewRecorder()
+	handler := handleAddPlayer(testCtx.handlerCtx)
+
+	testCtx.mockTemplate.EXPECT().Render(rr, nil)
+
+	handler.ServeHTTP(rr, req)
+
+	assertStatusOk(rr, t)
+}
+
+func makePostRequest(method, URL string, data map[string]string) (*http.Request, error) {
+	formData := url.Values{}
+	for k, v := range data {
+		formData.Set(k, v)
+	}
+	dataEcoded := bytes.NewBufferString(formData.Encode())
+	req, err := http.NewRequest(method, URL, dataEcoded)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded; param=value")
+	return req, nil
+}
+
+func TestHandleAddPlayerPostOk(t *testing.T) {
+	testCtx := makeTestCtx(t)
+	defer testCtx.mockCtrl.Finish()
+
+	playerName := "test"
+	data := map[string]string{"playerName": playerName}
+	req, err := makePostRequest("POST", "/players/add", data)
+	if err != nil {
+		t.Fatal(err)
+	}
+	rr := httptest.NewRecorder()
+	handler := handleAddPlayer(testCtx.handlerCtx)
+
+	testCtx.mockDB.EXPECT().CreatePlayer(playerName)
+
+	handler.ServeHTTP(rr, req)
+
+	if status := rr.Code; status != http.StatusFound {
+		t.Errorf("expected redirect but got %v", status)
+	}
+}
+
+func TestHandleAddPlayerPostNotOk(t *testing.T) {
```

Comment:
Is this test case dependent on TestHandleAddPlayerPostOk to pass? Where does the "test" player gets added to db the first time?

Reply:
This test does not dependent on TestHandleAddPlayerPostOk to pass. We are mocking db interface, player was never actually added. We emulating that player alredy exist with `.Return(nil, errors.New("exists"))`